knimin/lib/data_access.py

```diff
@@ -3496,19 +3498,29 @@ def extract_sample_plates(self, sample_plate_ids, email, robot, kit_lot,
                 "extraction_kit_lot", kit_lot)
             tool_id = self.get_or_create_property_option_id(
                 "extraction_tool", tool)
-            sql = """INSERT INTO pm.dna_plate (
-                        email, name, created_on, sample_plate_id,
-                        extraction_robot_id, extraction_kit_lot_id,
-                        extraction_tool_id, notes)
-                     VALUES (%s, (SELECT name
-                                  FROM pm.sample_plate WHERE
-                                  sample_plate_id = %s),
-                             now(), %s, %s, %s, %s, %s)
-                     RETURNING dna_plate_id"""
+            if not names:
+                sql = """INSERT INTO pm.dna_plate (
+                            email, name, created_on, sample_plate_id,
+                            extraction_robot_id, extraction_kit_lot_id,
+                            extraction_tool_id, notes)
+                         VALUES (%s, (SELECT name
+                                      FROM pm.sample_plate WHERE
+                                      sample_plate_id = %s),
+                                 now(), %s, %s, %s, %s, %s)
+                         RETURNING dna_plate_id"""
+            else:
+                sql = """INSERT INTO pm.dna_plate (
+                            email, name, created_on, sample_plate_id,
+                            extraction_robot_id, extraction_kit_lot_id,
+                            extraction_tool_id, notes)
+                         VALUES (%s, %s, now(), %s, %s, %s, %s, %s)
+                         RETURNING dna_plate_id"""
             dna_plates = []
-            for p_id in sample_plate_ids:
-                TRN.add(sql, [email, p_id, p_id, robot_id, kit_lot_id,
-                              tool_id, notes])
+            for p_id, name in zip_longest(sample_plate_ids, names,
```

Comment:
if `len(names) > len(sample_plate_ids)`, then a `None` will be used as the `p_id` placing a `null` in the database for the `sample_plate_id`. Is that correct? It looks like the following should be enforced:

```python
if names:
    if len(names) != len(sample_plate_ids):
        raise ValueError("...")
```